src/qililab/instruments/mini_circuits/attenuator.py

```diff
@@ -43,13 +43,19 @@ class StepAttenuatorSettings(Instrument.InstrumentSettings):
     settings: StepAttenuatorSettings
     device: MiniCircuitsDriver
 
-    @Instrument.CheckDeviceInitialized
     @Instrument.CheckParameterValueFloatOrInt
-    def setup(self, parameter: Parameter, value: float | str | bool, channel_id: int | None = None):
+    def setup(  # type: ignore
+        self,
+        parameter: Parameter,
+        value: float | str | bool,
+        channel_id: int | None = None,
+        port_id: str | None = None,
```

Comment:
why are we adding `port_id`?

Reply:
pylint/mypy were complaining, that the child's didn't follow the parent interfaces,

So I could add an ignore or fix it, I don't think that breaks anything, but if you think we shouldn't do it anyway let me know again!

Reply:
Yes, it's complaining because you have added it in `Instrument` as well. The question is why adding it in the first place.

Reply:
It was complaining before me doing any change Vyron, I didn't add more code because I felt like 😅

I just checked, and the reason was that `qblox_module`, `qblox_qrm`'s and `qblox_qcm`'s had that parameter in the `setup`, so pylint or mypy was complaining 🙃

Do you think I shouldn't expand the qblox_behaviour up to the parent then? I could place ignores in those qblox child instead than expanding it 👌

Reply:
Removed `port_id`, changing the implementation from `system_control` as suggested!